SaveManager.py

```diff
@@ -421,6 +436,124 @@ def view_saves():
+    print("\t\t0. Choose at any time to cancel the whole process.")
+    print("================================")
+    
+    # Get the interval for the auto backup
+    try:
+        interval = int(input("\n\tChoose backup interval, in minutes (1-60): "))
+    except ValueError:
+        promptEnter(f"Input was not a valid number. Auto backup cancelled.")
+        return
+    
+    # Cancel the auto backup if the interval is 0
+    if interval == 0:
+        promptEnter("Auto backup cancelled.")
+        return
+    
+    # Check if the interval is within the valid range
+    if interval < 1 or interval > 60:
+        promptEnter(f"'{interval}' is not a valid input. Auto backup cancelled.")
+        return
+    
+    user_input = input("\n\tDo you want to set a backup max? (recommended) [Y/N]: ")
+    if user_input == "0":
+        promptEnter("Auto backup cancelled.")
+        return
+    elif (user_input.lower() == "y"):
+        try:
+            user_input = int(input("\n\tEnter the maximum number of backups (10-100): "))
+        except ValueError:
+            promptEnter(f"Input was not a valid number. Auto backup cancelled.")
+            return
+        if user_input == 0:
+            promptEnter("Auto backup cancelled.")
+            return
+        elif user_input < 10 or user_input > 100:
+            promptEnter(f"'{user_input}' is not a valid input. Auto backup cancelled.")
+            return
+        set_max_backups = True
+        max_backups = user_input
+        print(f"\n\tMaximum number of backups set to {max_backups}. Excess will be culled on next auto backup.")
+    else:
+        set_max_backups = False
+        max_backups = 100 # set to 100, just in case
+        print("\n\tMax backups set to unlimited.")
+
+
+    with backup_lock:
+        backup_active = True
+        auto_backup_option_text = f"Stop Auto Backup [set to {interval} minutes]"
+    backup_thread = threading.Thread(target=backup_timer, args=(interval,))
+    backup_thread.start()
+    promptEnter(f"Auto backup started with interval of {interval} minutes.")
+    return
+
+# used in the auto_backup function
+def backup_timer(interval):
+    global backup_active, auto_backup_alert_text
+    elapsed_time = 0
+
+    # Calculate the interval in seconds
+    interval_seconds = interval * 60   # Convert interval to seconds
+    # interval_seconds = 5 # debug interval
+
+    while backup_active:
+        time.sleep(1)  # Thread sleeps for 1 second intervals to avoid busy-waiting
+        with backup_lock:
+            if not backup_active:  # Check if timer should still be running
+                break
+        
+        elapsed_time += 1
+        if elapsed_time >= interval_seconds:
+            # print(f"Auto backup created at {datetime.datetime.now()}") # Debug print statement
+            auto_save(display_prompt=False)
+            elapsed_time = 0  # Reset elapsed time after performing a backup
```

Comment:
I think we should only do backups if there have been modifications to the save file since the last backup. Otherwise, if you remain afk in the game, or otherwise somehow play for a long time without hitting new save points, or leave this program running while not playing the game, your backup folder will become full of the same 10 - 100 saves, and you may even fill up your hard drive if you turn the backup limit off.

We can check for changes in the save files by looking at the modification timestamp.

Reply:
Definitely!